### PR TITLE
Sync ds json data

### LINKED ISSUE #274: Changes in provisioning file are not updated in Datasource
**What happened**:
I updated apiHost in Provisioning file, and restarted grafana. After restarted plugin was still making api calls to old apiHost.

synthetic-monitoring settings (`<grafana-url>/api/plugins/grafana-synthetic-monitoring-app/settings`) was showing updated value of apiHost

after some investigation, I found that Synthetic Monitoring Datasource still has old value (check it at `<grafana-url>/api/datasources`)

**What you expected to happen**:
after updating provisioning file and restarting grafana, plugin should have updated the value apiHost

**How to reproduce it (as minimally and precisely as possible)**:
- install synthetic-monitoring
- setup and use it
- change provisioning file
- restart grafana
- changes in provisioning file should be reflated in in data-source.

**Workaround:**
- disable plugin
- delete Synthetic Monitoring Datasource
- enabled plugin
- go to home screen and click monitor your systems

now datasource should get values from provisioning file, you can verify it at `<grafana-url>/api/datasources`

### LINKED ISSUE #419: Provision the plugin with datasource UIDs instead of names
Renaming a stack in Cloud results in the SM plugin losing track of which prometheus and loki datasources it should be talking to. This happens because prom/loki datasources get renamed when the stack gets renames, but the provisioning that informs the plugin doesn't have the new names. The plugin identifies datasources by name, so things stop working. We should look into updating the plugin provisioning to use UIDs instead (if possible), it will be more stable and the SM plugin will continue to work without interruption across renames.

### PR DESCRIPTION
**Pull request by rdubrock**
Closes #419 
Closes #274 

This will leverage the `preload: true` plugin setting to update to sync provisioned data with the SM datasource whether a use navigates to the plugin or not. Currently if things change users have to delete the SM datasource and reinitialize the plugin.

This also checks to see whether dashboards have changed and updates them.

TODO: tests!

### CODE REVIEW

File: src/module.ts [hunk 1]
```diff
@@ -1,11 +1,61 @@
...
+  ) {
+    return true;
+  }
+  return false;
+};
+
+const preloadInit = async () => {
+  const smDatasources = findSMDataSources();
+  const pluginSettings = await getPluginSettings();
+  // if there is no SM datasource, we have nothing to do. If there is more than 1 we don't know what to do.
```
Comment (jewbetcha) [inline src/module.ts:42]: Could maybe short circuit this function earlier based on what comes back in `smDatasources`

**Review by jewbetcha — COMMENTED** [review]
Any luck showing feedback to user if an update is happening?

**Comment by rdubrock** [comment]
> Any luck showing feedback to user if an update is happening?

I haven't experimented with that yet, will try next week I think. They do get a `Datasource updated` toast, which isn't super helpful. There's nothing for when dashboards update.

File: src/module.ts [hunk 2]
```diff
@@ -1,11 +1,61 @@
...
+import { getDashboardsNeedingUpdate, importAllDashboards } from 'dashboards/loader';
 
 export const plugin = new AppPlugin<GlobalSettings>().setRootPage(App).addConfigPage({
   title: 'Config',
   icon: 'cog',
   body: ConfigPage,
   id: 'config',
 });
+
+const getPluginSettings = async () =>
```
Comment (rdubrock) [inline src/module.ts:17]: We have to fetch this from the backend because `plugin.meta` is undefined at this point in the execution

**Comment by rdubrock** [comment]
So we _can_ give a toast notifying the user that the SM dashboards were updated. I skipped adding one for when the datasource is updated as that's an implementation detail and should be transparent to users. I think this is a pretty high risk change, if folks are willing to give this a careful 👀 and maybe even test locally I'd appreciate it.

**Pull request closed without merging**